src/message/header/mailbox.rs

```diff
@@ -306,15 +306,12 @@ mod test {
     #[test]
     fn parse_multi_with_name_containing_comma() {
         let from: Vec<Mailbox> = vec![
-            "Test, test <1@example.com>".parse().unwrap(),
-            "Test2, test2 <2@example.com>".parse().unwrap(),
```

Comment:
These are invalid `mailbox`es and therefore an invalid mailbox list. They should be quoted (as lettre does). This updates the code to use lettre's encoder which properly formats the values. This now more directly tests the problem faced by the original bug (https://github.com/lettre/lettre/issues/690) by testing inserting the address usign lettre's usual mechanisms.